libvast/src/system/exporter.cpp

```diff
@@ -73,9 +73,12 @@ void attach_stream(exporter_actor::stateful_pointer<exporter_state> self) {
         [continuous](const stream_state& state) {
           if (continuous)
             return false;
-          return state.self_ptr->state.query_status.received
-                   == state.self_ptr->state.query_status.expected
-                 && state.self_ptr->state.results.empty();
+          auto should_end = state.self_ptr->state.query_status.received
+                              == state.self_ptr->state.query_status.expected
+                            && state.self_ptr->state.results.empty();
+          if (should_end)
+            shutdown_stream(state.self_ptr->state.source);
```

Comment:
Based on the CAF header files documentation this shouldn't cause any weird issues.

Should we wait on neverlord's comment before merging it? @dominiklohmann

Reply:
I think we can merge this as-is. The reply will help regardless in improving our understanding.